mpi_ring_c_test.sh

```diff
@@ -0,0 +1,80 @@
+#!/bin/bash
+
+function check_efa_ompi {
+    out=$1
+    grep -q "mtl:ofi:prov: efa" $out
+    if [ $? -ne 0 ]; then
+        echo "efa provider not used with Open MPI"
+        exit 1
+    fi
+}
+
+function check_efa_impi {
+    out=$1
+    grep -q "libfabric provider: efa" $out
```

Comment:
This seems like a pretty generic phrase in order to check Intel MPI. I don't feel that great about it, but no alternative suggested.